ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_CppSamples/ZipHelper.h

```diff
@@ -0,0 +1,129 @@
+#include <QString>
+#include <QStringList>
+
+// minizip-ng headers
+#include <minizip-ng/zip.h>
+#include <minizip-ng/unzip.h>
+
+//--------------------------------------------------------------------
+
+class ZipHelper : public QObject
+{
+    Q_OBJECT
+
+public:
+    enum class Result
+    {
+        EndOfListOfFile     = UNZ_END_OF_LIST_OF_FILE,
+        ErrNo               = UNZ_ERRNO,
+        EndOfFile           = UNZ_EOF,
+        ParameterError      = UNZ_PARAMERROR,
+        BadZipFile          = UNZ_BADZIPFILE,
+        InternalError       = UNZ_INTERNALERROR,
+        CrcError            = UNZ_CRCERROR
+    };
+    Q_ENUM(Result)
+
+public:
+    Q_INVOKABLE bool extractAll(const QString &outputPath);
+
+    void setPath(const QString& path);
+
+signals:
+    void pathChanged();
+
+    void extractError(const QString& fileName, const QString& outputFileName, ZipHelper::Result result);
+    void extractProgress(const QString& fileName, const QString& outputFileName, qreal percent);
+    void extractCompleted();
+
+public:
+    explicit ZipHelper(QObject* parent = nullptr);
+    ZipHelper(const QString& path, QObject* parent = nullptr);
+    ~ZipHelper();
+
+private:
+    bool zrOpen(const QString& filePath);
+    void zrClose();
+
+    unzFile zrHandle() const { return m_unzFile; }
+    bool zrIsOpen() const { return m_unzFile != nullptr; }
+
+    bool extractAll(QDir& outputDir);
+    bool extractCurrentFile(const QString& outputFilePath);
+
+    QString currentFileName();
+    QDateTime currentFileDateTime();
+    unz_file_info64 currentFileInfo();
+
+    bool gotoFirstFile();
+    bool gotoNextFile();
+
+// FileUtilty methods needed for ZipHelper::setPath to work
+//--------------------------------------------------------------------
+
+public:
+  enum class PathVariableType
+  {
+    NotFound            = 0,
+    ApplicationProperty = 1,
+    Settings            = 2,
+    Environment         = 3
+  };
+private:
+  static bool setDateTime(const QString& filePath, const QDateTime& accessed, const QDateTime& modified);
+  static QString resolvedPath(const QString& path, bool useFallbacks = true);
+  static QString pathVariable(const QString& name, const QString& defaultValue = QString(), bool useFallbacks = true);
+  static void setPathVariable(const QString& name, const QString& value);
+
+  static PathVariableType pathVariableType(const QString& name, bool useFallbacks = true);
+  static bool pathVariableExists(const QString& name, bool useFallbacks = true);
+
+  static QString defaultPathVariable(const QString& name);
+  static QString defaultHomePath();
+  static QString defaultTempPath();
+private:
+  static QByteArray pathPropertyName(const QString& name);
+
+//--------------------------------------------------------------------
+
+private:
+    void zwClose(const QString& globalComment = QString());
+
+private:
+    bool exists() const;
+    bool zrOpen();
+
+    const QString& path() const { return m_Path; }
+
+private:
```

Comment:
I recommend removing the redundant `private` sections. There are five in a row. We just need the first one.

I realize this is old code, but we should tidy up things that are low risk.